### PR TITLE
DEV: Update plugin to match latest guidelines

### PR DESCRIPTION
**Pull request by nbianca**
This commit updates the plugin to the latest guidelines, as shown in discourse-plugin-skeleton, which involves moving a lot of the code to dedicated files, use proper namespaces, and use the autoloader as much as possible, etc.

### CODE REVIEW

**Comment by jjaffeux** [comment]
Can you describe what you changed to not break anymore please?

**Comment by nbianca** [comment]
I renamed app/core_ext/plugin_instance.rb → lib/plugin/instance.rb. The name of the old file did not follow the rules of the autoloader and that caused issues.

**Comment by davidtaylorhq** [comment]
Hmm, is this definitely working as expected in both dev and production mode? I would've guessed that Zeitwerk will be confused if there are multiple files which define `lib/plugin/instance.rb` (i.e. one in core, and one in discourse-automation). How will it know which one to open?

I feel like monkey-patches like this might be better as a rails initializer, wrapped by `Rails.application.reloader.to_prepare` (like our core [freedom patches initializer](https://github.com/discourse/discourse/blob/8f1a5c93929465daa89dca8cebd972475b6181d3/config/initializers/002-freedom_patches.rb#L8-L8)), or alternatively wrapped in a `reloadable_patch` block in `plugin.rb`

**Comment by nbianca** [comment]
It seems to be working fine and I tried that by spinning up a production site using discourse_docker. I will have a look at your suggestions. :mag:

**Comment by nbianca** [comment]
I do not think that would work because there are other plugins that are calling that method in after_initialize which implies some dependencies between plugins.

File: app/controllers/discourse_automation/admin_automations_controller.rb [hunk 1]
```diff
@@ -1,7 +1,7 @@
 # frozen_string_literal: true
 
 module DiscourseAutomation
-  class AdminDiscourseAutomationAutomationsController < ::ApplicationController
```
Comment (CvX) [inline app/controllers/discourse_automation/admin_automations_controller.rb:4]: Those controllers were previously "public"?

**Review by CvX — APPROVED** [review]

File: app/controllers/discourse_automation/admin_automations_controller.rb (hunk 1 above)
Comment (nbianca) [reply app/controllers/discourse_automation/admin_automations_controller.rb:4]: Unfortunately yes.

Comment (davidtaylorhq) [reply app/controllers/discourse_automation/admin_automations_controller.rb:4]: Looks like it had an AdminConstraint on the route, so I think it was safe

https://github.com/discourse/discourse-automation/blob/d17a5aa26f67d25042686fe7ed54bff6d9bb63a3/plugin.rb#L101-L103

Comment (nbianca) [reply app/controllers/discourse_automation/admin_automations_controller.rb:4]: Oh right. I have seen that and then it totally slipped my mind. 🤦🏻‍♀️

**Pull request merged**